nixos/modules/services/web-apps/lubelogger.nix

```diff
@@ -0,0 +1,94 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+with lib;
+
+let
+  cfg = config.services.lubelogger;
+in
+{
+  options = {
+    services.lubelogger = {
+      enable = mkEnableOption "Lubelogger, a self-hosted, open-source, web-based vehicle maintenance and fuel milage tracker";
+
+      package = mkPackageOption pkgs "lubelogger" { };
+
+      dataDir = mkOption {
+        description = "Path to Lubelogger config and metadata inside of /var/lib.";
+        default = "lubelogger";
+        type = types.str;
+      };
+
+      port = mkOption {
+        description = "The TCP port Lubelogger will listen on.";
+        default = 5000;
+        readOnly = true; # Lubelogger does not allow you to configure the port it runs on
```

Comment:
To customize Lubelogger, I think you can use the environment variable:
```nix
        Kestrel__Endpoints__Http__Url = "http://localhost:5003";
```

Reply:
can confirm this works, at least on aarch64-darwin